apps/server/src/modules/lesson/controller/dto/lesson.response.ts

```diff
@@ -70,7 +71,13 @@ export class LessonResponse {
 		description: 'The id of the Course entity',
 		pattern: '[a-f0-9]{24}',
 	})
-	courseId: EntityId;
+	courseId?: EntityId;
```

Comment:
same as above, why is the courseId optional?

Reply:
according to database, we have either courseId or courseGroupId, but not both.
I think the entity should also be changed to reflect this....

Reply:
agree, will you take that with you to create arch ticket for it?

Reply:
out of scope here, because the change touches stuff like the file service
I will discuss / create followup ticket